components/merkledb/src/access/mod.rs

```diff
@@ -143,6 +135,14 @@ pub enum AccessErrorKind {
         actual: IndexType,
     },
 
+    /// Index has invalid name.
+    #[fail(display = "{}", _0)]
+    InvalidIndexName(String),
+
+    /// System index has invalid name.
+    #[fail(display = "{}", _0)]
+    InvalidSystemIndexName(String),
```

Comment:
This variant should correspond to the case when client code attempts to access a system view / index (e.g., "\_\_METADATA\_\_"). Correspondingly, it should not contain an error message, but rather be an empty variant.